### PR TITLE
improve query builder launch and save interactions

### PR DESCRIPTION
**Pull request by akphi**
## Summary

Closes https://github.com/finos/legend-studio/issues/631

Also,

- [x] Fix styling for several element creation modals to be more consistent
- [x] Apply `mode` strategy for `QueryBuilderState` (similar strategy we adopted in `EditorStoreMode` in #642 
- [x] Minor improvements and bug fixes for setting up query from `DataSpace`:
  - [x] Fix bug with cannot go to dataspace project
  - [x] Allow getting dataspaces from latest snapshots
  - [x] Fix a bug where the indicator in data space search box showing showing target project instead of the project that the dataspace belongs to

## How did you test this change?

- [ ] Test(s) added

I added screen recordings

#### Legend Query new look n feel 💪 
https://user-images.githubusercontent.com/13574879/141489855-b29c19ba-2f8e-46e1-a5af-3eb6422e8a66.mov

#### Studio mapping editor new look n feel 🎸 
https://user-images.githubusercontent.com/13574879/141490064-49fdf4e9-9e89-417a-9f9b-7e75f2a86a58.mov

### CODE REVIEW

File: packages/legend-extension-dsl-data-space/src/components/query/DataSpaceQuerySetup.tsx [hunk 1]
```diff
@@ -178,6 +186,19 @@ export const DataspaceQuerySetup = observer(
...
             />
+            <button
+              className={clsx('query-setup__data-space__use-snapshot-btn', {
+                'query-setup__data-space__use-snapshot-btn--active':
+                  querySetupState.toGetSnapShot,
+              })}
+              tabIndex={-1}
+              title={`[${
+                querySetupState.toGetSnapShot ? 'on' : 'off'
+              }] Toggle show data spaces from snapshot releases instead of latest releases`}
```
[the remark this answers is not in the document]
Comment (akphi) [reply packages/legend-extension-dsl-data-space/src/components/query/DataSpaceQuerySetup.tsx:197]: seeing both is confusing don't you think? I want to keep `SNAPSHOT` to something a little bit obscure as it's more meant for development

File: packages/legend-extension-dsl-data-space/src/models/protocols/pure/DSLDataSpace_PureProtocolProcessorPlugin.ts [hunk 2]
```diff
@@ -319,6 +321,13 @@ export class ResolvedDataSpaceExecutionContext {
  * element pointers to actual reference, hence this model.
  */
 export class ResolvedDataSpace {
+  taggedValues: {
```
[the remark this answers is not in the document]
Comment (akphi) [reply packages/legend-extension-dsl-data-space/src/models/protocols/pure/DSLDataSpace_PureProtocolProcessorPlugin.ts:324]: want to add a few things like `uuid` to it, also I don't want to leak `V1_...` out into the component.

File: packages/legend-server-depot/src/DepotServerClient.ts [hunk 3]
```diff
@@ -92,6 +93,7 @@ export class DepotServerClient extends AbstractServerClient {
     classifierPath: string,
     options?: {
       search?: string | undefined;
+      scope?: DepotScope | undefined;
```
[the remark this answers is not in the document]
Comment (akphi) [reply packages/legend-server-depot/src/DepotServerClient.ts:96]: hm, I don't have a strong comeback for that, but this is how `depot` APIs are kind of naturally set up. In the older infrastructure of `depot` before it gets open-sourced, this is also the pattern of usage they follow

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -22,6 +22,8 @@
 
 ## How did you test this change?
 
+- [ ] Test(s) added
```
[the remark this answers is not in the document]
Comment (akphi) [reply .github/PULL_REQUEST_TEMPLATE.md:25]: Make sense, I will do this in the next PR

File: .changeset/chilled-tomatoes-know.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+'@finos/legend-extension-dsl-data-space': minor
+---
+
+Add new panels to data space viewer: a placeholder for test coverage (`WIP`) and a panel to view tagged values and stereotypes.
```
[the remark this answers is not in the document]
Comment (akphi) [reply .changeset/chilled-tomatoes-know.md:5]: 😢

**Review by MauricioUyaguari — APPROVED** [review]

**Pull request merged**